task/net/src/server_basic.rs

```diff
@@ -8,282 +8,43 @@
 
 use drv_stm32h7_eth as eth;
 
+use crate::bsp_support;
+use crate::generated;
+use crate::server::{DeviceExt, GenServerImpl, Storage};
 use core::cell::Cell;
-use idol_runtime::{ClientError, NotificationHandler, RequestError};
 use mutable_statics::mutable_statics;
-use smoltcp::iface::{Interface, Neighbor, SocketHandle, SocketStorage};
-use smoltcp::socket::UdpSocket;
-use smoltcp::wire::{
-    EthernetAddress, IpAddress, IpCidr, Ipv6Address, Ipv6Cidr,
-};
-use task_net_api::{
-    LargePayloadBehavior, RecvError, SendError, SocketName, UdpMetadata,
-};
-use userlib::{sys_post, sys_refresh_task_id};
-
-use crate::generated::{self, SOCKET_COUNT};
-use crate::server::NetServer;
-use crate::{idl, ETH_IRQ, NEIGHBORS, WAKE_IRQ_BIT};
-
-type NeighborStorage = Option<(IpAddress, Neighbor)>;
+use smoltcp::wire::{EthernetAddress, Ipv6Address};
+use task_net_api::UdpMetadata;
 
 /// Grabs references to the server storage arrays.  Can only be called once!
-pub fn claim_server_storage_statics() -> (
-    &'static mut [NeighborStorage; NEIGHBORS],
-    &'static mut [SocketStorage<'static>; SOCKET_COUNT],
-    &'static mut [IpCidr; 1],
-) {
+fn claim_server_storage_statics() -> &'static mut [Storage; 1] {
```

Comment:
I attempted to make this generic, i.e. `fn claim_server_storage_statics<const N: usize>()`, but ran into issues with the capture:
```
error[E0401]: can't use generic parameters from outer function
   --> task/net/src/server.rs:583:39
    |
581 | fn claim_server_storage_statics<const N: usize>() -> &'static mut [Storage; N] {
    |                                       - const parameter from outer function
582 |     mutable_statics! {
583 |         static mut STORAGE: [Storage; N] = [|| Default::default(); _];
    |                                       ^ use of generic parameter from outer function
```

Too bad...

Reply:
Yeah, statics are not cloned once per monomorphization of their containing generic function, unfortunately.